Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/nbclient/client.py", line 1093, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/nbclient/client.py", line 559, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/nbclient/client.py", line 854, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/nbclient/client.py", line 756, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
result.visualize()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mFileNotFoundError[0m                         Traceback (most recent call last)
[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphviz/backend/execute.py[0m in [0;36mrun_check[0;34m(cmd, input_lines, encoding, capture_output, quiet, **kwargs)[0m
[1;32m     82[0m             [0;32massert[0m [0miter[0m[0;34m([0m[0minput_lines[0m[0;34m)[0m [0;32mis[0m [0minput_lines[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 83[0;31m             [0mproc[0m [0;34m=[0m [0m_run_input_lines[0m[0;34m([0m[0mcmd[0m[0;34m,[0m [0minput_lines[0m[0;34m,[0m [0mkwargs[0m[0;34m=[0m[0mkwargs[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     84[0m         [0;32melse[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphviz/backend/execute.py[0m in [0;36m_run_input_lines[0;34m(cmd, input_lines, kwargs)[0m
[1;32m    102[0m [0;32mdef[0m [0m_run_input_lines[0m[0;34m([0m[0mcmd[0m[0;34m,[0m [0minput_lines[0m[0;34m,[0m [0;34m*[0m[0;34m,[0m [0mkwargs[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 103[0;31m     [0mpopen[0m [0;34m=[0m [0msubprocess[0m[0;34m.[0m[0mPopen[0m[0;34m([0m[0mcmd[0m[0;34m,[0m [0mstdin[0m[0;34m=[0m[0msubprocess[0m[0;34m.[0m[0mPIPE[0m[0;34m,[0m [0;34m**[0m[0mkwargs[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    104[0m [0;34m[0m[0m

[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/subprocess.py[0m in [0;36m__init__[0;34m(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, encoding, errors, text)[0m
[1;32m    857[0m [0;34m[0m[0m
[0;32m--> 858[0;31m             self._execute_child(args, executable, preexec_fn, close_fds,
[0m[1;32m    859[0m                                 [0mpass_fds[0m[0;34m,[0m [0mcwd[0m[0;34m,[0m [0menv[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m

[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/subprocess.py[0m in [0;36m_execute_child[0;34m(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, start_new_session)[0m
[1;32m   1703[0m                         [0merr_msg[0m [0;34m=[0m [0mos[0m[0;34m.[0m[0mstrerror[0m[0;34m([0m[0merrno_num[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m-> 1704[0;31m                     [0;32mraise[0m [0mchild_exception_type[0m[0;34m([0m[0merrno_num[0m[0;34m,[0m [0merr_msg[0m[0;34m,[0m [0merr_filename[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   1705[0m                 [0;32mraise[0m [0mchild_exception_type[0m[0;34m([0m[0merr_msg[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;31mFileNotFoundError[0m: [Errno 2] No such file or directory: PosixPath('dot')

The above exception was the direct cause of the following exception:

[0;31mExecutableNotFound[0m                        Traceback (most recent call last)
[0;32m/tmp/ipykernel_2134/503654577.py[0m in [0;36m<module>[0;34m[0m
[0;32m----> 1[0;31m [0mresult[0m[0;34m.[0m[0mvisualize[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m
[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/dask/base.py[0m in [0;36mvisualize[0;34m(self, filename, format, optimize_graph, **kwargs)[0m
[1;32m    212[0m         [0mhttps[0m[0;34m:[0m[0;34m//[0m[0mdocs[0m[0;34m.[0m[0mdask[0m[0;34m.[0m[0morg[0m[0;34m/[0m[0men[0m[0;34m/[0m[0mlatest[0m[0;34m/[0m[0moptimize[0m[0;34m.[0m[0mhtml[0m[0;34m[0m[0;34m[0m[0m
[1;32m    213[0m         """
[0;32m--> 214[0;31m         return visualize(
[0m[1;32m    215[0m             [0mself[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m    216[0m             [0mfilename[0m[0;34m=[0m[0mfilename[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m

[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/dask/base.py[0m in [0;36mvisualize[0;34m(*args, **kwargs)[0m
[1;32m    743[0m         [0;32mraise[0m [0mNotImplementedError[0m[0;34m([0m[0;34m"Unknown value color=%s"[0m [0;34m%[0m [0mcolor[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m    744[0m [0;34m[0m[0m
[0;32m--> 745[0;31m     [0;32mreturn[0m [0mdot_graph[0m[0;34m([0m[0mdsk[0m[0;34m,[0m [0mfilename[0m[0;34m=[0m[0mfilename[0m[0;34m,[0m [0;34m**[0m[0mkwargs[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    746[0m [0;34m[0m[0m
[1;32m    747[0m [0;34m[0m[0m

[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/dask/dot.py[0m in [0;36mdot_graph[0;34m(dsk, filename, format, **kwargs)[0m
[1;32m    269[0m     """
[1;32m    270[0m     [0mg[0m [0;34m=[0m [0mto_graphviz[0m[0;34m([0m[0mdsk[0m[0;34m,[0m [0;34m**[0m[0mkwargs[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 271[0;31m     [0;32mreturn[0m [0mgraphviz_to_file[0m[0;34m([0m[0mg[0m[0;34m,[0m [0mfilename[0m[0;34m,[0m [0mformat[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    272[0m [0;34m[0m[0m
[1;32m    273[0m [0;34m[0m[0m

[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/dask/dot.py[0m in [0;36mgraphviz_to_file[0;34m(g, filename, format)[0m
[1;32m    286[0m         [0mformat[0m [0;34m=[0m [0;34m"png"[0m[0;34m[0m[0;34m[0m[0m
[1;32m    287[0m [0;34m[0m[0m
[0;32m--> 288[0;31m     [0mdata[0m [0;34m=[0m [0mg[0m[0;34m.[0m[0mpipe[0m[0;34m([0m[0mformat[0m[0;34m=[0m[0mformat[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    289[0m     [0;32mif[0m [0;32mnot[0m [0mdata[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m    290[0m         raise RuntimeError(

[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphviz/piping.py[0m in [0;36mpipe[0;34m(self, format, renderer, formatter, quiet, engine, encoding)[0m
[1;32m    114[0m             [0mraw[0m [0;34m=[0m [0mself[0m[0;34m.[0m[0m_pipe_lines[0m[0;34m([0m[0;34m*[0m[0margs[0m[0;34m,[0m [0minput_encoding[0m[0;34m=[0m[0mself[0m[0;34m.[0m[0m_encoding[0m[0;34m,[0m [0;34m**[0m[0mkwargs[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m    115[0m             [0;32mreturn[0m [0mraw[0m[0;34m.[0m[0mdecode[0m[0;34m([0m[0mencoding[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 116[0;31m         [0;32mreturn[0m [0mself[0m[0;34m.[0m[0m_pipe_lines[0m[0;34m([0m[0;34m*[0m[0margs[0m[0;34m,[0m [0minput_encoding[0m[0;34m=[0m[0mself[0m[0;34m.[0m[0m_encoding[0m[0;34m,[0m [0;34m**[0m[0mkwargs[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m
[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphviz/backend/piping.py[0m in [0;36mpipe_lines[0;34m(engine, format, input_lines, input_encoding, renderer, formatter, quiet)[0m
[1;32m    147[0m     [0mkwargs[0m [0;34m=[0m [0;34m{[0m[0;34m'input_lines'[0m[0;34m:[0m [0;34m([0m[0mline[0m[0;34m.[0m[0mencode[0m[0;34m([0m[0minput_encoding[0m[0;34m)[0m [0;32mfor[0m [0mline[0m [0;32min[0m [0minput_lines[0m[0;34m)[0m[0;34m}[0m[0;34m[0m[0;34m[0m[0m
[1;32m    148[0m [0;34m[0m[0m
[0;32m--> 149[0;31m     [0mproc[0m [0;34m=[0m [0mexecute[0m[0;34m.[0m[0mrun_check[0m[0;34m([0m[0mcmd[0m[0;34m,[0m [0mcapture_output[0m[0;34m=[0m[0;32mTrue[0m[0;34m,[0m [0mquiet[0m[0;34m=[0m[0mquiet[0m[0;34m,[0m [0;34m**[0m[0mkwargs[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    150[0m     [0;32mreturn[0m [0mproc[0m[0;34m.[0m[0mstdout[0m[0;34m[0m[0;34m[0m[0m
[1;32m    151[0m [0;34m[0m[0m

[0;32m/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphviz/backend/execute.py[0m in [0;36mrun_check[0;34m(cmd, input_lines, encoding, capture_output, quiet, **kwargs)[0m
[1;32m     86[0m     [0;32mexcept[0m [0mOSError[0m [0;32mas[0m [0me[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m     87[0m         [0;32mif[0m [0me[0m[0;34m.[0m[0merrno[0m [0;34m==[0m [0merrno[0m[0;34m.[0m[0mENOENT[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 88[0;31m             [0;32mraise[0m [0mExecutableNotFound[0m[0;34m([0m[0mcmd[0m[0;34m)[0m [0;32mfrom[0m [0me[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     89[0m         [0;32mraise[0m[0;34m[0m[0;34m[0m[0m
[1;32m     90[0m [0;34m[0m[0m

[0;31mExecutableNotFound[0m: failed to execute PosixPath('dot'), make sure the Graphviz executables are on your systems' PATH
ExecutableNotFound: failed to execute PosixPath('dot'), make sure the Graphviz executables are on your systems' PATH

